lib/pgBackRest/Db.pm

```diff
@@ -90,7 +92,12 @@ sub new
 
         if (!isDbLocal({iRemoteIdx => $self->{iRemoteIdx}}))
         {
-            $self->{oProtocol} = protocolGet(CFGOPTVAL_REMOTE_TYPE_DB, $self->{iRemoteIdx});
+            eval
+            {
+                $self->{oProtocol} = protocolGet(CFGOPTVAL_REMOTE_TYPE_DB, $self->{iRemoteIdx}, {bWarnOnError => $bWarnOnError});
+                return true;
+            }
+            or do {};
         }
```

Comment:
If the protocol object is undefined ->connect will connect to any server that happens to be running on the local host.  This is probably not what you intend.